test/distributed/org/apache/cassandra/distributed/impl/AbstractCluster.java

```diff
@@ -1095,7 +1096,27 @@ public void close()
                            .filter(i -> !i.isShutdown())
                            .map(IInstance::shutdown)
                            .collect(Collectors.toList());
-        FBUtilities.waitOnFutures(futures,1L, TimeUnit.MINUTES);
+        try
+        {
+            FBUtilities.waitOnFutures(futures,1L, TimeUnit.MINUTES);
+        }
+        catch (RuntimeException t)
+        {
+            if (t.getCause() instanceof TimeoutException)
```

Comment:
a timeout is really not useful as you don't have enough information to know what happened... this will try to show what threads are left so you *could* figure out what is blocked.  This has been so helpful to find the threads that are blocking!

Reply:
Super useful! Great idea.

Reply:
I think I broke this when I improved the error msg... ill try to fix